actions/cloudbolt_plugins/configure_nodes_for_rke/cb_plugin_1570640510857138_mjMOkye.py

```diff
@@ -0,0 +1,233 @@
+    :param blueprint_context:
+    :return: iterable (yield) over Server objects
+    """
+    for key, value in blueprint_context.items():
+        if isinstance(value, dict) and 'servers' in value:
+            server_query_set = value.get('servers')
+            for server_obj in server_query_set:
+                yield server_obj
+
+
+def prepare_server_hosts(user, blueprint_context, ssh_public_key):
+    docker_script = 'yum -y install docker || exit 1;\n' + \
+                    'systemctl enable docker || exit 1;\n' + \
+                    'useradd {};\n'.format(user) + \
+                    'groupadd docker\n' + \
+                    'usermod -aG docker {} || exit 1\n'.format(user) + \
+                    'mkdir -p /home/{}/.ssh || exit 1;\n'.format(user) + \
+                    'echo \'{}\' >> /home/{}/.ssh/authorized_keys || exit 1;\n'.format(ssh_public_key, user) + \
+                    'chown -R {} /home/{}/.ssh || exit 1;\n'.format(user, user) + \
+                    'chmod 755 /home/{}/.ssh || exit 1;\n'.format(user) + \
+                    'chmod 644 /home/{}/.ssh/authorized_keys || exit 1;\n'.format(user) + \
+                    'echo \'net.ipv6.conf.all.forwarding=1\' >> /etc/sysctl.conf || exit 1\n' + \
+                    'sysctl -p /etc/sysctl.conf || exit 1'
+    for tcp_port_num in [443, 10250, 2379, 2380, 6443]:
+        docker_script += '\nfirewall-offline-cmd --add-port={}/tcp || exit 1;'.format(tcp_port_num)
+    for udp_port_num in [8285, 8472]:
+        docker_script += '\nfirewall-offline-cmd --add-port={}/udp || exit 1;'.format(udp_port_num)
+    docker_script += '\nsystemctl restart firewalld;\n'
+    logger.info(f"docker script:\n{docker_script}")
+    for server in find_all_servers(blueprint_context=blueprint_context):
+        set_progress("Starting script execution on server {}".format(server.ip))
+        server.execute_script(script_contents=docker_script, timeout=700)
+        server.reboot()
+    set_progress("Waiting for server(s) to begin reboot.")
+    time.sleep(10)
+    for server in find_all_servers(blueprint_context=blueprint_context):
+        server.wait_for_os_readiness()
+
+
+def kubernetes_up(cluster_yml_name: str):
+    cmd = f"{RKEDIR}/rke up --config={RKEDIR}/{cluster_yml_name}"
+    output = run_command(cmd)
+    set_progress(f"RKE up finished!. {output}")
+
+
+def run(_job, *_args, **kwargs):
+    """
+    main entry point for the plugin
+    """
+    user = 'cbrke'
+    blueprint_context = kwargs.get('blueprint_context', {})
+    ssh_public_key, ssh_private_key = create_ssh_keypair()
+    prepare_server_hosts(user, blueprint_context, ssh_public_key)
+
+    os.makedirs(RKEDIR, exist_ok=True)
+    with open(f"{RKEDIR}/rke_private_key.pem", 'w') as fl:
+        fl.write(ssh_private_key)
+    with open(f"{RKEDIR}/rke_public_key.pem", 'w') as fl:
+        fl.write(ssh_public_key)
+
+    ips = find_all_server_ips(kwargs.get('blueprint_context', {}))
```

Comment:
```suggestion
    ips = find_all_server_ips(blueprint_context)
```
I _think_ we can make that change.

Reply:
Any word on this change?

Reply:
I tried this and exception was thrown. will need more investigation.